tests/common/plugins/ptfadapter/__init__.py

```diff
@@ -63,5 +96,11 @@ def ptfadapter(ptfhost, testbed):
     ptfhost.command('supervisorctl reread')
     ptfhost.command('supervisorctl update')
 
+    # Force a restart of ptf_nn_agent to ensure that it is in good status.
+    ptfhost.command('supervisorctl restart ptf_nn_agent')
```

Comment:
Yes, it could be. While testing this change, I noticed that the same script passed after ptf_nn_agent is restarted. This should be a good workaround for issue #2032 before it is fully fixed in ptf_nn_agent.